src/tile.js

```diff
@@ -75,31 +74,39 @@ export default class Tile {
             { tile: this.buildAsMessage() })
         .then(message => {
             scene.buildTileCompleted(message);
+        }).catch(error => {
+            throw error;
         });
     }
 
     // Process geometry for tile - called by web worker
     // Returns a set of tile keys that should be sent to the main thread (so that we can minimize data exchange between worker and main thread)
     static buildGeometry (tile, layers, rules, styles) {
-        var name, layer, feature, style, feature_style;
+        var feature, style, feature_style;
         var vertex_data = {};
         var style_vertex_data;
 
-        tile.debug.rendering = +new Date();
```

Comment:
It's actually more important to me to track these metrics at a tile level. We could just track the tile, or track the sources and sum them at the end of this method.